From a8cf85668b8c858d28ac6fc54156d07b417a6888 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Wed, 16 Dec 2020 18:40:55 +0800
Subject: [PATCH] Change IllegalStateException to QuietException and explain
 that is a plugin


diff --git a/proxy/src/main/java/net/md_5/bungee/UserConnection.java b/proxy/src/main/java/net/md_5/bungee/UserConnection.java
index 96df0175..30b848f7 100644
--- a/proxy/src/main/java/net/md_5/bungee/UserConnection.java
+++ b/proxy/src/main/java/net/md_5/bungee/UserConnection.java
@@ -310,7 +310,8 @@ public final class UserConnection implements ProxiedPlayer
 
             if ( getServer() == null && !ch.isClosing() )
             {
-                throw new IllegalStateException( "Cancelled ServerConnectEvent with no server or disconnect." );
+                // FlameCord - Change IllegalStateException to QuietException and explain that is a plugin
+                throw new QuietException("A plugin cancelled ServerConnectEvent with no server or disconnect.");
             }
             return;
         }
-- 
2.31.1.windows.1

